src/main/java/benben/Deadline.java

```diff
@@ -0,0 +1,82 @@
+package benben;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * A type of task that has a specific date to be completed
+ */
+public class Deadline extends Task {
+    /**
+     * The deadline by which the task should be completed
+     */
+    protected LocalDate ddl;
+
+    /**
+     * The Formatter sued to parse the string into a LocalDateTime
+     */
+    protected final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+    /**
+     * Instantiates a new Deadline.
+     *
+     * @param description the description of the task
+     * @param ddl         the ddl by which it should be completed
+     * @throws BenBenException if the deadline cannot be parsed
+     */
+    public Deadline(String description, String ddl) throws BenBenException{
+        super(description);
+        try {
+            this.ddl = LocalDate.parse(ddl);
+        } catch (DateTimeParseException e) {
+            throw new BenBenException("The date and time is of the wrong format! Please use yyyy-MM-dd");
+        }
+
+    }
+    @Override
+    public String toString() {
+        return "[D] " + super.toString() + " (by: " + getDdl() + ")";
```

Comment:
for instances like this (as well as below), would using a String.format method work better for readability?